### ISSUE TITLE
Help with WRF+SWAN compilation.

### ISSUE THREAD

**Issue by losufba** [jcwarner-usgs/COAWST#282]
Dear John W. and other forum colleagues.

I have a problem with COAWST version 3.4 of my server. Before you ask, why am I using such an old version? Because it is stable for my libraries and the current versions of COAWST don't work for many reasons. So I was wondering if you could help me with this error.

I can compile and run COAWST with WRF+ROMS, ROMS+SWAN, but when I go to compile WRF+SWAN I get an error that I haven't been able to resolve.

Below I attach my compilation outputs and the attached .f90 file where I'm having the problem.

f951: Warning: Nonexistent include directory ‘.././Build’ [-Wmissing-include-dirs]
atm_coupler.f90:567:32:

  567 |       IF (MOD(num_steps+offset, nATM2WAV(1,1)).eq.0) THEN
      |                                1
Error: Function ‘natm2wav’ at (1) has no IMPLICIT type
atm_coupler.f90:578:32:

  578 |       IF (MOD(num_steps+offset, nATMFWAV(1,1)).eq.0) THEN
      |                                1
Error: Function ‘natmfwav’ at (1) has no IMPLICIT type
Command exited with non-zero status 1
0.29user 0.04system 0:00.34elapsed 99%CPU (0avgtext+0avgdata 46588maxresident)k
0inputs+8outputs (0major+11830minor)pagefaults 0swaps
make[3]: [../configure.wrf:361: atm_coupler.o] Error 1 (ignored)

[compile_output.txt](https://github.com/user-attachments/files/16192902/compile_output.txt)

Best Regards

**Comment by jcwarner-usgs** [comment]
usually, we build a few files first into the Build dir, files for MCT and the coupling.  One of those files would have the variable "natm2wav" in it.  But for some reason your system did not build those files first, and there is no Build dir, so it can not be found.

If you build roms + wrf, what does that build look like?
I cant spend time to work on that older version,  if you want to let me know of the problems with the newer version perhaps i can help.
-j

**Comment by losufba** [comment]
Good morning John, 

Yes, when I build WRF+ROMS or ROMS+SWAN, the files are created in the build normally. But for some reason, I don't know what the problem is, it's WRF+SWAN. In 12 years of working with COAWST, I've never had this problem. Have you ever seen this problem?

Well, as you know, the biggest problem with the new versions is the instability in compiling WRF. I tend to use stable versions for me, like 3.4 and 3.7. However, when I encountered this problem, I tested the current version 3.8 and I can't compile WRF either as serial or as dmpar.

I'm going to do some tests now in the afternoon (Brazil time) and I'll send you some points of error of the latest version.

Thanks for your attention.

Best Regards

**Comment by losufba** [comment]
Good morning John, how are you?

I'm sorry, but I haven't managed to install version 3.8 yet, but I've tried to adjust a few points in the WRF+SWAN compilation.

I noticed that the main error is in the atm_coupler module, according to the error below. There are even a few other errors associated with this module. Do you think I can compile it on my own?

Here's the error: 

rm -f atm_coupler.o
sed -e "s/^\!.*'.*//" -e "s/^ *\!.*'.*//" atm_coupler.F > atm_coupler.G
/usr/bin/cpp -P -traditional -I/home/camobi/COAWST/WRF/inc -DEM_CORE=1 -DNMM_CORE=0 -DNMM_MAX_DIM=2600 -DDA_CORE=0 -DWRFPLUS=0 -DIWORDSIZE=4 -DDWORDSIZE=8 -DRWORDSIZE=4 -DLWORDSIZE=4 -DNONSTANDARD_SYSTEM_SUBR  -DWRF_USE_CLM -DUSE_NETCDF4_FEATURES -DWRFIO_NCD_LARGE_FILE_SUPPORT  -DDM_PARALLEL -DNETCDF -DUSE_ALLOCATABLES -Dwrfmodel -DGRIB1 -DINTIO -DKEEP_INT_AROUND -DLIMIT_ARGS -DBUILD_RRTMG_FAST=1 -DCONFIG_BUF_LEN=65536 -DMAX_DOMAINS_F=21 -DMAX_HISTORY=25 -DNMM_NEST=0  -I. -traditional-cpp -DUSE_NETCDF4_FEATURES -DWRFIO_NCD_LARGE_FILE_SUPPORT  -DJOE_TC -D'PROJECT_HEADER="/home/camobi/COAWST/Projects/COAWST/bts_coupled_2014//joe_tc.h"' -D'GLOBAL_DEFS="/home/camobi/COAWST//ROMS/Include/globaldefs.h"' -DEM_CORE=1 -DNMM_CORE=0 -DNMM_MAX_DIM=2600 -DDA_CORE=0 -DWRFPLUS=0 -DIWORDSIZE=4 -DDWORDSIZE=8 -DRWORDSIZE=4 -DLWORDSIZE=4 -DNONSTANDARD_SYSTEM_SUBR  -DWRF_USE_CLM -DUSE_NETCDF4_FEATURES -DWRFIO_NCD_LARGE_FILE_SUPPORT  -DDM_PARALLEL -DNETCDF -DUSE_ALLOCATABLES -Dwrfmodel -DGRIB1 -DINTIO -DKEEP_INT_AROUND -DLIMIT_ARGS -DBUILD_RRTMG_FAST=1 -DCONFIG_BUF_LEN=65536 -DMAX_DOMAINS_F=21 -DMAX_HISTORY=25 -DNMM_NEST=0  -I. -traditional-cpp -DUSE_NETCDF4_FEATURES -DWRFIO_NCD_LARGE_FILE_SUPPORT   atm_coupler.G  > atm_coupler.bb
/home/camobi/COAWST/WRF/tools/standard.exe atm_coupler.bb | /usr/bin/cpp -P -traditional -traditional-cpp -DUSE_NETCDF4_FEATURES -DWRFIO_NCD_LARGE_FILE_SUPPORT > atm_coupler.f90
rm -f atm_coupler.G atm_coupler.bb
time mpif90  -o atm_coupler.o -c -O2 -ftree-vectorize -funroll-loops -w -ffree-form -ffree-line-length-none -fconvert=big-endian -frecord-marker=4   -I../dyn_em -I../dyn_nmm  -I/home/camobi/COAWST/WRF/external/esmf_time_f90  -I/home/camobi/COAWST/WRF/main -I/home/camobi/COAWST/WRF/external/io_netcdf -I/home/camobi/COAWST/WRF/external/io_int -I/home/camobi/COAWST/WRF/frame -I/home/camobi/COAWST/WRF/share -I/home/camobi/COAWST/WRF/phys -I/home/camobi/COAWST/WRF/wrftladj -I/home/camobi/COAWST/WRF/chem -I/home/camobi/COAWST/WRF/inc -I/usr/local/include -I.././Build  -I/usr/local/mct/include   atm_coupler.f90
f951: Warning: Nonexistent include directory ‘.././Build’ [-Wmissing-include-dirs]
atm_coupler.f90:686:32:

  686 |       IF (MOD(num_steps+offset, nATM2WAV(1,1)).eq.0) THEN
      |                                1
Error: Function ‘natm2wav’ at (1) has no IMPLICIT type
atm_coupler.f90:697:32:

  697 |       IF (MOD(num_steps+offset, nATMFWAV(1,1)).eq.0) THEN
      |                                1
Error: Function ‘natmfwav’ at (1) has no IMPLICIT type
Command exited with non-zero status 1
0.48user 0.04system 0:00.55elapsed 97%CPU (0avgtext+0avgdata 65368maxresident)k
32inputs+8outputs (0major+18089minor)pagefaults 0swaps
make[3]: [../configure.wrf:361: atm_coupler.o] Error 1 (ignored)

**Comment by jcwarner-usgs** [comment]
really need to see the full build log.  if wrf gets an error, it will keep going.  so you have to look at the full log and start from the top.

**Comment by losufba** [comment]
Dear John,

I decided to follow your suggestion and try running COAWST in version 3.8 and no longer in version 3.4. However, as I told you earlier, I'm having a lot of problems compiling WRF in this version.

I attach the compilation output, after running build_coawst.sh with WRF in gfortran in dmpar mode.

I'm using: 
openmpi 4.1.4
zlib-1.2.11
hdf5-1.14.0
netcdf-c-4.9.0
netcdf-fortran-4.6.0

I hope you can help me.
[output_compile.txt](https://github.com/user-attachments/files/16203840/output_compile.txt)

**Comment by jcwarner-usgs** [comment]
there are some strange things going on in that build.
can you try to build wrf+roms and send that full build?


-j

**Comment by losufba** [comment]
john, how are you?

I apologize for the delay, I've been trying different ways out of my problem. I decided to format my server and start with an ifort compile and impi. 

Things have improved, but now I have a problem with mct. I've had this problem before and it was solved with a COAWST update. I hope you can help me.

I've attached all my files so you can check.
[compilation_output.txt](https://github.com/user-attachments/files/16368639/compilation_output.txt)
[build_coawst.sh.txt](https://github.com/user-attachments/files/16368647/build_coawst.sh.txt)

best regards

**Comment by jcwarner-usgs** [comment]
the error is actually near the top of the output

Serial C compiler (mostly for tool generation):
which SCC
make[1]: *** [Makefile:64: configcheck] Erro 1
make[1]: a sair da pasta "/home/camobi/COAWST/WRF"

It is having trouble to find SCC.
this is trying to build WRF.  you need to use MPI, not serial.
so you need to select a different config number.  something with mpi.

**Comment by losufba** [comment]
Good evening John, how are you my friend?

I apologize for the delay, but I've been working more than 12 hours a day on COAWST to run it on my institutional server. 

First of all, I'd like to say that I succeeded. I took several prints in fortran codes until I came up with some solutions.

I'm really having a lot of problems with the new versions of COAWST. I tested the current version on 2 other servers and got the same errors. I've updated libraries with limited success, although I've made a lot of progress.

I decided to go back to my stable version of COAWST 3.4 with GCC, G++ and GFORTRAN 9. It really is very good. I was having trouble compiling WRF+SWAN and discovered that I had to add the nopie command to configure.wrf first, as it was causing problems. 
FCFLAGS += -no-pie 
LDFLAGS += -no-pie

After some personal adjustments on my server, I compiled WRF+ROMS and then compiled WRF+ROMS+SWAN with -nocleanwrf. And it worked...

Thank you very much for your support and willingness to help. You are an extinct programmer helper.
